### PR TITLE
Setting instead of hardcoded model.

### PR DESCRIPTION
**Pull request by nowackipawel**

### CODE REVIEW

**Comment by jozefrebjak** [comment]
@nowackipawel Great idea. I would also like to see something similar with `LoginController`. Now we need to extend Controllers if we want to add logic for example based on groups within `getLoginRedirect($user)`function.

**Comment by MGatner** [comment]
@nowackipawel Could you please update the Config `Auth::$userProvider` to be typed to the user provider interface? That way we will be sure these calls are using something we can anticipate.

**Comment by nowackipawel** [comment]
```
     * You can change it as long as they adhere to the
     * Sparks\Shield\Interfaces\UserProvider.```

**Comment by MGatner** [comment]
Yes, but also provide the type in the docblock. Since this is a string we can't really enforce it via PHP but static analysis will catch failures. Basically change the bottom of **src/Config/Auth.php** to be this:
```php

    /**
     * --------------------------------------------------------------------
     * User Provider
     * --------------------------------------------------------------------
     * The name of the class that handles user persistence.
     * By default, this is the included UserModel, which
     * works with any of the database engines supported by CodeIgniter.
     * 
     * @var class-string<\Sparks\Shield\Interfaces\UserProvider>
     */
    public $userProvider = 'Sparks\Shield\Models\UserModel';
```

(Or include a `use` statement at top and use the class basename).

**Comment by nowackipawel** [comment]
@MGatner , I've never been an doc expert. Thx.

File: src/Controllers/RegisterController.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function registerAction()
      */
     protected function getUserProvider()
     {
-        return model('UserModel'); // @phpstan-ignore-line
+        return model(setting('Auth.userProvider')); // @phpstan-ignore-line
```
Comment (lonnie-vault) [inline src/Controllers/RegisterController.php:93]:
```suggestion
        return model(setting('Auth.userProvider')); 
```

Comment (lonnieezell) [inline src/Controllers/RegisterController.php:93]:
```suggestion
        return model(setting('Auth.userProvider')); 
```

**Comment by lonnieezell** [comment]
please run `composer style` on the code to fix the style issues and commit those changes.

**Review by lonnieezell — APPROVED** [review]

**Review by MGatner — APPROVED** [review]

**Pull request merged**